src/router/summary_router.py

```
@@ -0,0 +1,28 @@
+_lock = threading.Lock()  # Lock 추가
+
+def get_summarizer():
+    global _summarizer
+    if _summarizer is None:
+        with _lock:  # Thread-safe 처리
+            if _summarizer is None:
+                _summarizer = MeetingSummarizer()
+    return _summarizer
+
+@router.post("/summarize", response_model=SummaryResponse)
+async def summarize_meeting(summarizer: MeetingSummarizer = Depends(get_summarizer)):
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)